src/os/bluestore/BlueStore.cc

```diff
@@ -10400,14 +10675,54 @@ int BlueStore::_fsck_on_open(BlueStore::FSCKDepth depth, bool repair)
 
   bluefs_used_blocks = used_blocks;
 
+  if (bdev_label_multi) {
+    // Forcibly mark regions of bdev label clones as used.
+    // If an object happens to be using it we will get an error and a repair applied.
+    // We can move away data only if it was allocated for object in BlueStore,
+    // we are unable to move away BlueFS data.
+
+    // skip first bdev label in this check
+    for (uint64_t position : bdev_labels_broken) {
+      uint64_t length = std::max<uint64_t>(BDEV_LABEL_BLOCK_SIZE, alloc_size);
+      bool is_taken_by_bluefs = false;
+      apply_for_bitset_range(position, length, alloc_size, bluefs_used_blocks,
+        [&](uint64_t pos, mempool_dynamic_bitset& bs) {
+          is_taken_by_bluefs |= bs.test_set(pos);
+        }
+      );
+      if (is_taken_by_bluefs) {
+        // We are unable to fix it.
+        dout(1) << "fsck bdev label at 0x" << std::hex << position << std::dec
+                <<  "taken by bluefs, cannot be fixed" << dendl;
+      } else {
+        if (repair) {
+          // Mark blocks so we could move offending objects away.
+          bdev_labels_in_repair.push_back(position);
+        }
+      }
+    }
+    // Mark bits or locations of all bdev labels.
+    for (size_t i = 0; i < bdev_label_positions.size(); i++) {
```

Comment:
shouldn't this go over bdev_labels_in_repair instead?

Reply:
No, we can have bdev labels locations that have proper data, but are in collision with some object.

Reply:
well, the above comment is a bit confusing then - in fact you mark all the possible bdev locations here. Irrespective of their bluefs usage.

Reply:
I have completely different comment here. Lets check after push.